File: backend/src/main/java/com/example/backend/entity/Transaction.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "transactions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String transactionType; // Payment, Refund, Partial Refund
    private String transactionStatus; // Success, Failed, Pending
    private Double transactionAmount;
    private String gatewayResponse;

    private LocalDateTime transactionDate = LocalDateTime.now();

    @ManyToOne
    @JoinColumn(name = "payment_id")
    private Payment payment;

    @Column(nullable = false)
    private Boolean isActive = true;
}

File: frontend/src/app/shared/models/transaction.ts
export type TransactionType = 'PAYMENT' | 'REFUND' | 'PARTIAL_REFUND';
export type TransactionStatus = 'SUCCESS' | 'FAILED' | 'PENDING';

export interface Transaction {
  id: number;
  paymentId: number;
  transactionType: TransactionType;
  transactionStatus: TransactionStatus;
  transactionAmount: number;
  transactionDate: Date;
  gatewayResponse?: string;
  isActive: boolean;
}
